core/DataAccess/ArchiveSelector.php

```diff
@@ -61,80 +61,41 @@ public static function getArchiveIdAndVisits(ArchiveProcessor\Parameters $params
 
         $numericTable = ArchiveTableCreator::getNumericTable($dateStart);
 
-        $minDatetimeIsoArchiveProcessedUTC = null;
-        if ($minDatetimeArchiveProcessedUTC) {
-            $minDatetimeIsoArchiveProcessedUTC = Date::factory($minDatetimeArchiveProcessedUTC)->getDatetime();
-        }
-
         $requestedPlugin = $params->getRequestedPlugin();
         $segment         = $params->getSegment();
         $plugins = array("VisitsSummary", $requestedPlugin);
 
         $doneFlags      = Rules::getDoneFlags($plugins, $segment);
         $doneFlagValues = Rules::getSelectableDoneFlagValues($includeInvalidated, $params);
 
-        $results = self::getModel()->getArchiveIdAndVisits($numericTable, $idSite, $period, $dateStartIso, $dateEndIso, $minDatetimeIsoArchiveProcessedUTC, $doneFlags, $doneFlagValues);
-
-        if (empty($results)) {
-            return false;
+        $results = self::getModel()->getArchiveIdAndVisits($numericTable, $idSite, $period, $dateStartIso, $dateEndIso, null, $doneFlags);
+        if (empty($results)) { // no archive found
+            return [false, false, false, false];
         }
 
-        $idArchive = self::getMostRecentIdArchiveFromResults($segment, $requestedPlugin, $results);
-
-        $idArchiveVisitsSummary = self::getMostRecentIdArchiveFromResults($segment, "VisitsSummary", $results);
-
-        list($visits, $visitsConverted) = self::getVisitsMetricsFromResults($idArchive, $idArchiveVisitsSummary, $results);
-
-        if (false === $visits && false === $idArchive) {
-            return false;
+        $result = self::findArchiveDataWithLatestTsArchived($results, $requestedPlugin, $segment);
+        if (!isset($result['idarchive'])
+            || !isset($result['nb_visits'])
+        ) {
+            return [false, false, false, true];
```

Comment:
btw here it returned for me `isAnyArchiveExists = true` even though there was no archive with a done flag. Only an archive with a `nb_visits`. This can happen if eg the archiving fails before the done flag was written but after nb_visits was written. Not sure it's important.